src/lib.rs

```diff
@@ -11,13 +13,41 @@ pub const SSH_EOF: c_int = -127;
 pub enum ssh_session_struct {}
 pub type ssh_session = *mut ssh_session_struct;
 
+pub type ssh_auth_e = c_int;
+pub const SSH_AUTH_SUCCESS: ssh_auth_e = 0;
+pub const SSH_AUTH_DENIED: ssh_auth_e = 1;
+pub const SSH_AUTH_PARTIAL: ssh_auth_e = 2;
+pub const SSH_AUTH_INFO: ssh_auth_e = 3;
+pub const SSH_AUTH_AGAIN: ssh_auth_e = 4;
+pub const SSH_AUTH_ERROR: ssh_auth_e = -1;
+
+pub enum ssh_key_struct {}
+pub type ssh_key = *mut ssh_key_struct;
+
+pub type ssh_known_hosts_e = c_int;
+pub const SSH_KNOWN_HOSTS_ERROR: ssh_known_hosts_e = -2;
+pub const SSH_KNOWN_HOSTS_NOT_FOUND: ssh_known_hosts_e = -1;
+pub const SSH_KNOWN_HOSTS_UNKNOWN: ssh_known_hosts_e = 0;
+pub const SSH_KNOWN_HOSTS_OK: ssh_known_hosts_e = 1;
+pub const SSH_KNOWN_HOSTS_CHANGED: ssh_known_hosts_e = 2;
+pub const SSH_KNOWN_HOSTS_OTHER: ssh_known_hosts_e = 3;
+
 extern "C" {
     pub fn ssh_new() -> ssh_session;
     pub fn ssh_connect(session: ssh_session) -> c_int;
     pub fn ssh_disconnect(session: ssh_session) -> c_int;
     pub fn ssh_free(session: ssh_session);
 
     pub fn ssh_get_error(error: *mut c_void) -> *const c_char;
+    pub fn ssh_get_server_publickey(session: ssh_session, key: *mut ssh_key) -> c_int;
```

Comment:
Can you move this and the `ssh_key` definitions down with "Public key related functions"?